File: domain/src/main/java/com/example/message/MessageService.java
package com.example.message;

import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;

@Slf4j
@Service
public class MessageService {

    public void send() throws InterruptedException {
        Thread.sleep(500);
        log.info("예매가 성공적으로 되었습니다.");
    }

}

File: module-application/src/main/kotlin/com/example/redis/message/in/MessageService.kt
package com.example.redis.message.`in`

import org.springframework.scheduling.annotation.Async
import org.springframework.stereotype.Service
import org.springframework.transaction.event.TransactionPhase
import org.springframework.transaction.event.TransactionalEventListener

@Service
class MessageService: MessageUseCase {

    override fun send() {
        Thread.sleep(500)
        println("예약 발송 성공")
    }
}